projects/observability/src/shared/components/top-n/top-n-chart.component.test.ts

```diff
@@ -1,12 +1,18 @@
-import { createHostFactory, Spectator } from '@ngneat/spectator/jest';
+import { CommonModule } from '@angular/common';
+import { FormattingModule, NavigationService } from '@hypertrace/common';
+import { TooltipModule } from '@hypertrace/components';
+import { getMockFlexLayoutProviders } from '@hypertrace/test-utils';
+import { createHostFactory, mockProvider, Spectator } from '@ngneat/spectator/jest';
 import { TopNChartComponent, TopNData } from './top-n-chart.component';
 
-describe('TopN Chart Component', () => {
+describe('Top N Chart Component', () => {
   let spectator: Spectator<TopNChartComponent>;
 
   const createHost = createHostFactory({
     component: TopNChartComponent,
-    shallow: true
+    shallow: true,
+    imports: [CommonModule, TooltipModule, FormattingModule],
+    providers: [mockProvider(NavigationService), ...getMockFlexLayoutProviders()]
```

Comment:
I think we can get rid of `CommonModule`, `TooltipModule` imports and `getMockFlexLayoutProviders` provider here

Reply:
I doesn't work without it. May be i will have to mock the directive. But I am not sure I want to do a shallow test here. Testing the display value post formatting seems an important test case for this component. So I think i must test it against the actual formatting pipe. I am not testing the internals of display formatter, only using it as a black box because the correct output is relevant to my top-n-view. @aaron-steinfeld What are your thoughts?

Reply:
Generally, the job of this test ends once it goes into the next component/pipe - the formatting it produces should be tested by that pipe. There is a place for deeper tests especially for complex components working together, but I don't think a display component like this is one of them.

Reply:
True, but this is a very basic component. I don't think I should skip checking whether the correct text (value with percentage) is being displayed or not.

Reply:
Removing the imports and mock providers breaks the test. Should i ng-mock the tooltip and display pipe? Or is there a better way? I want this change in soon.

Reply:
@anandtiwary It passes for me. That's why I didn't ask you to remove `FormattingModule` import and get rid of everything else

```
const createHost = createHostFactory({
    component: TopNChartComponent,
    shallow: true,
    imports: [FormattingModule],
});
```

Reply:
pipes have to be mocked or provided, but a shallow test means you don't have to mock components or directives unless you're interacting with them in the test. @skjindal93 's suggestion seems right